superset-frontend/src/components/Label/index.tsx

```diff
@@ -71,49 +75,57 @@ export default function Label(props: LabelProps) {
     info,
   } = colors;
 
-  let backgroundColor = grayscale.light3;
-  let backgroundColorHover = onClick ? primary.light2 : grayscale.light3;
-  let borderColor = onClick ? grayscale.light2 : 'transparent';
-  let borderColorHover = onClick ? primary.light1 : 'transparent';
-  let color = grayscale.dark1;
+  let baseColor;
+  if (type === 'default') {
+    baseColor = grayscale;
+  } else if (type === 'success') {
+    baseColor = success;
+  } else if (type === 'alert') {
+    baseColor = alert;
+  } else if (type === 'warning') {
+    baseColor = warning;
+  } else if (type === 'danger') {
+    baseColor = error;
+  } else if (type === 'info') {
+    baseColor = info;
+  } else if (type === 'secondary') {
+    baseColor = secondary;
+  } else {
+    baseColor = primary;
+  }
+  let color = baseColor.dark2;
+  let borderColor = baseColor.light1;
+  let backgroundColor = baseColor.light2;
+  let backgroundColorHover = onClick ? baseColor.light1 : backgroundColor;
+  let borderColorHover = onClick ? baseColor.base : borderColor;
 
-  if (type !== 'default') {
-    color = grayscale.light4;
+  // TODO - REMOVE IF BLOCK LOGIC WHEN shades are fixed to be aligned in terms of brightness
+  // currently shades for >=light2 are not aligned for primary, default and secondary
+  if (['default', 'primary', 'secondary'].includes(type)) {
```

Comment:
@kgabryje I resorted to add custom logic here instead of trying to align the theme. The larger theming effort should fix all of this and remove the need from having ANY color-related logic in the component itself.

Reply:
Alright looks good for now 👍